services/image/src/routes/image.ts

```diff
@@ -0,0 +1,42 @@
+import { Hono } from 'hono'
+import { cors } from 'hono/cors'
+import { allowedOrigin } from '@kodadot/workers-utils'
+import { Env } from '../utils/constants'
+import { getImageByPath, uploadCFI } from '../utils/cloudflare-images'
+import { vValidator } from '@hono/valibot-validator'
+import Hash from 'ipfs-only-hash'
+
+import {
+  UploadImage,
+  uploadImageRequestSchema,
+} from '../schemas/uploadImageRequestSchema'
+
+const app = new Hono<{ Bindings: Env }>()
+
+app.use('/*', cors({ origin: allowedOrigin }))
+
+app.post('/upload', vValidator('form', uploadImageRequestSchema), async (c) => {
+  const { file } = await c.req.parseBody<UploadImage>()
+
+  const path = await Hash.of(new Uint8Array(await file.arrayBuffer()))
```

Comment:
nit: actually I'm good with `crypto.randomUUID()`. this package seems old

Reply:
Wont it break the schema of cfi?

Reply:
Since this endpoint is for general images to put on Cloudflare Images, and we store the URL on the profile database, it would not conflict

Reply:
This means it always overrides the image after every upload. I'm worried about the security side. Is it possible that another user changed our profile image?

Reply:
Let's go back to using `crypto.randomUUID()` for now,

Reply:
> This means it always overrides the image after every upload. I'm worried about the security side. Is it possible that another user changed our profile image?

Please check - https://github.com/kodadot/private-workers/issues/180#issuecomment-2175788357

User should provide a signature

Reply:
> Please check - https://github.com/kodadot/private-workers/issues/180#issuecomment-2175788357

wooah, cool. We can go with the user address, then.

We can probably get rid of the `image` column also: https://github.com/kodadot/private-workers/blob/34071cea5d09be98b24b8e9a685e9da56a205d19/services/profile/schema.ts#L9

---

Just want to make sure. If we use the user address, do we need to pass it here? https://github.com/kodadot/nft-gallery/pull/10434/files#diff-a73ed45becdc3ec15de43f950f7232fa7b7268c7a5aca4c35a32d6a1c3acad7eR44-R60. attacker still able to hijack the body value isn't?

Reply:
> wooah, cool. We can go with the user address, then.

so we do something like ? `{address}_image` and `{address}_banner` 

where `image` and `banner` is passed in the request as `type` or something like that

> Just want to make sure. If we use the user address, do we need to pass it here? 

we can just pass the the signed message and grab the address from there, can't we ?

Reply:
> so we do something like ? `{address}_image` and `{address}_banner`

yes, something like this

> also are just using the `SUB` address ? , what would happen if the user edited his profile using first `EVM` and then `SUB` , we would have 2 images from the same user

I guess so. Since we store each address, it means the user has a separate name/description also on each chain


> we can just pass the the signed message and grab the address in the worker , can't we ?

yes, unless we add verify also in the worker. Anyway, using the `address` format is closely coupled with the `profile`. I think putting the endpoint in the `profile-worker` is better

Reply:
So let's make this  PR `${address}_image` and `${adress}_banner` then let's do opsec in followup

Reply:
done

Reply:
> And the address is not verified by this worker. I can hijack it, tho. I can show it to you guys if you want

yep , you can change the image of other users without updating their profile

didn't do this bcz of viki's [comment ](https://github.com/kodadot/workers/pull/300#discussion_r1646130132)

> How about this one? I prefer to put the endpoint to profile-worker

+1 ,  will wait for https://github.com/kodadot/private-workers/pull/182

Reply:
Let's close this one and create a new endpoint for the profile worker instead. If we go with this one, the attacker can update some user profiles without verifying the signature